python/ray/data/_internal/dataset_logger.py

```diff
@@ -43,27 +43,18 @@ def _initialize_logger(self) -> logging.Logger:
         for writing to the Dataset log file. Not intended (nor should it be necessary)
         to call explicitly. Assumes that `ray.init()` has already been called prior
         to calling this method; otherwise raises a `ValueError`."""
-        # In the case where logger has not yet been set up in another file,
-        # initialize basic configs to set log level and format.
-        logging.basicConfig(
-            level=LOGGER_LEVEL.upper(),
-            format=LOGGER_FORMAT,
-        )
-
-        # We initialize a logger using the given base `log_name`, which
-        # logs to stdout. Logging with this logger to stdout is enabled by the
-        # `log_to_stdout` parameter in `self.get_logger()`.
         stdout_logger = logging.getLogger(self.log_name)
+
+        # Add stderr handler.
+        formatter = logging.Formatter(fmt=LOGGER_FORMAT)
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+        handler.setLevel(LOGGER_LEVEL.upper())
+        stdout_logger.addHandler(handler)
+        stdout_logger.propagate = False
```

Comment:
I don't think `stdout_logger` is ever used at all in the code base. `get_logger()` returns the file logger, not this one. There is a comment above that says that the `DatasetLogger` will log to stdout if `log_to_stdout=True`, but this is I think happening because the file logger propagates the message up to the `root` logger, which is configured by `basicConfig`.

Reply:
ah yeah it's for keeping the backward compatibility for now because we want to minimize the failure radius here. We can make it use the default formating once we fix this issue.